lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -940,18 +940,41 @@ def test_user_with_passing_existing_generating_cert(self):
         self.assertIn("Certificate is already being created.", resp.content)
 
     @patch('courseware.grades.grade', Mock(return_value={'grade': 'Pass', 'percent': 0.75}))
+    @override_settings(CERT_QUEUE='certificates', SEGMENT_IO_LMS_KEY="foobar", FEATURES={'SEGMENT_IO_LMS': True})
     def test_user_with_passing_existing_downloadable_cert(self):
-        # If user has passing grade but also has existing downloadable cert
-        # then json will return cert generating message with bad request code
+        # If user has already downloadable certificate then he can again re-generate the
+        # the cert.
         GeneratedCertificateFactory.create(
             user=self.student,
             course_id=self.course.id,
             status=CertificateStatuses.downloadable,
             mode='verified'
         )
-        resp = self.client.post(self.url)
-        self.assertEqual(resp.status_code, HttpResponseBadRequest.status_code)
-        self.assertIn("Certificate has already been created.", resp.content)
+
+        analytics_patcher = patch('courseware.views.analytics')
+        mock_tracker = analytics_patcher.start()
+        self.addCleanup(analytics_patcher.stop)
+
+        with patch('capa.xqueue_interface.XQueueInterface.send_to_queue') as mock_send_to_queue:
+            mock_send_to_queue.return_value = (0, "Successfully queued")
```

Comment:
@srpearce student will not see this message. sorry i did't mention this earlier.